[WNMGDS-1984] Blog page and article for Release 5.0 (#2233)
CMSgov/design-system: packages/docs/content/blog/index.mdx

---
title: What's new
---

import Video from '../../src/components/content/Video';
import scssToCssImg from '../../src/images/scss-to-css.svg';

## Release 5.0

<time datetime="2022-11-01" className="ds-text-body--sm">November 1, 2022</time>

Hey everyone, we're doing something new! We've saved up a program increment's worth of updates into a single release, and it's going out with a little fanfare. Watch or listen to our introduction video below—complete with closed captions—and check out our other demo videos sprinkled throughout this article. Enjoy!

<div className="ds-u-measure--wide ds-u-margin-top--2">
  <Video name="release-5.0" />
</div>

### What's new in 5.0

#### Component maturity checklist

> We established a checklist that measures each component's path toward maturity and stability through [well-defined and repeatable processes](https://github.com/CMSgov/design-system/blob/pwolfert/blog/COMPONENT_MATURITY.md). This _maturity model_ evaluates a component's accessibility, code, Sketch assets, and use of design tokens as a series checklist items that have a complete, incomplete, or not-applicable status. This checklist helps us track progress on components and also helps application teams make informed decisions about when to use certain components, when to wait, or when to contribute.

Reviewer: ```suggestion
We established a checklist that measures each component's path toward maturity and stability through [well-defined and repeatable processes](https://github.com/CMSgov/design-system/blob/pwolfert/blog/COMPONENT_MATURITY.md). This _maturity model_ evaluates a component's accessibility, code, Sketch assets, and use of design tokens as a series of checklist items that have a complete, incomplete, or not-applicable status. This checklist helps us track progress on components and also helps application teams make informed decisions about when to use certain components, when to wait, or when to contribute.
```

<div className="ds-u-measure--wide ds-u-margin-top--2">
  <Video name="component-maturity-demo" />
</div>

#### Single-column layout

> We implemented a single-column layout design pattern for Helathcare.gov that teams can use to align their page layouts in both design and [code](https://design.cms.gov/layouts/Healthcare/one-col-page-layout?theme=healthcare). The single page layout has a space for a header, content, and a footer.  We are looking into expanding upon this page layout design patterns to offer more page layouts for Healthcare and Medicare.

Reviewer: ```suggestion
We implemented a single-column layout design pattern for Helathcare.gov that teams can use to align their page layouts in both design and [code](https://design.cms.gov/layouts/Healthcare/one-col-page-layout?theme=healthcare). The single-page layout has a space for a header, content, and a footer.  We are looking into expanding upon page layout design patterns to offer more page layouts for Healthcare and Medicare.
```

Author: Hyphen is correct gramatically

Author: I'll remove the `this`

<div className="ds-u-measure--wide ds-u-margin-top--2">
  <Video name="page-layout-demo" />
</div>

#### New dialog

> The Modal Dialog and Drawer components are now using the new [HTML `<dialog>` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog). There are no visual changes associated with this change, but it's a major tune-up under the hood. Before the advent of a specific `<dialog>` element in HTML, web developers had to create the modal dialog experience themselves by dressing up non-semantic elements with custom CSS and JavaScript. This update will improve accessibility and remove third-party dependencies that we used to have to rely on. You can read more about the [research behind the change in github discussions](https://github.com/CMSgov/design-system/discussions/1831).

Reviewer: ```suggestion
The Modal Dialog and Drawer components are now using the new [HTML `<dialog>` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog). There are no visual changes associated with this change, but it's a major tune-up under the hood. Before the advent of a specific `<dialog>` element in HTML, web developers had to create the modal dialog experience themselves by dressing up non-semantic elements with custom CSS and JavaScript. This update will improve accessibility and remove third-party dependencies that we used to have to rely on. You can read more about the [research behind the change in GitHub discussions](https://github.com/CMSgov/design-system/discussions/1831).
```

- Rebuilt our visual-regression-testing suite to test all three themes of our design system on multiple browsers at multiple screen sizes. 

Mamma mia, those are some thorough checks! All these things help us be more confident that we're providing quality components.

#### Sketch alert symbol update 

We have consolidated and updated our alert symbol to take it from multiple alert symbols to a single alert symbol with optional overrides. The override panel supports standard and lightweight designs, changing the alert icon, and showing and hiding the alert heading and body. The simplified approach to the alert and the practice of utilizing layer styles more to achieve design flexibility is something we are going to continue weaving into other sketch symbols as we create and update them. Finally, this updated alert symbol now can easily adapt to small screen sizes without breaking the design of the alert.

<div className="ds-u-measure--wide ds-u-margin-top--2">
  <Video name="sketch-alert-demo" />
</div>

#### Camilo Sketch plug-in discovery 

> [We have been researching options](https://github.com/CMSgov/design-system/discussions/2231) for making our Sketch UI-kits easier to maintain and scale. Currently, we maintain three UI-kits for Healthcare, Medicare, and the Core Design System. Each time there is a change needed we have to make that change to the three different UI-kits. This can be time consuming and creates challenges when trying to replicate the exact symbol in different Sketch UI-kits. One of the more promising solutions to solve this problem is a plugin for Sketch called Camilo. It applies a theme to a single UI-kit so you can have multiple brands use the same UI-kit. The Camilo plugin has some issues that prevent it from being a solid solution for the CMS Design System. We are continuing to do some final explorations before closing the door on this option.

Reviewer: ```suggestion
[We have been researching options](https://github.com/CMSgov/design-system/discussions/2231) for making our Sketch UI kits easier to maintain and scale. Currently, we maintain three UI kits for Healthcare, Medicare, and the Core Design System. Each time there is a change needed we have to make that change to the three different UI kits. This can be time-consuming and creates challenges when trying to replicate the exact symbol in different Sketch UI kits. One of the more promising solutions to solve this problem is a plugin for Sketch called Camilo. It applies a theme to a single UI kit so you can have multiple brands use the same UI kit. The Camilo plugin has some issues that prevent it from being a solid solution for the CMS Design System. We are continuing to do some final explorations before closing the door on this option.
```

#### Feedback & discussion page

> We are using [Github Discussions](https://github.com/CMSgov/design-system/discussions) to discuss topics pertaining to the Design System to promote more conversation and communication within pages of the Design System. Your feedback and input helps pave the way for how the Design System grows and changes.

Reviewer: ```suggestion
We are using [GitHub Discussions](https://github.com/CMSgov/design-system/discussions) to discuss topics about the Design System to promote more conversation and communication within pages of the Design System. Your feedback and input help pave the way for how the Design System grows and changes.
```

### What's on deck

#### Sass/CSS changes

Product teams have experienced pain points consuming our CSS as Sass, so we explored alternative options for distributing our CSS. Sass has been the recommended method of importing Sass from the beginning of the design system, but we've run into serious obstacles with upgrading to newer versions of Sass. 

<div className="ds-u-measure--wide ds-u-text-align--center ds-u-margin-y--3">
  <img src={scssToCssImg} alt="Sass to CSS" className="ds-u-display--inline-block" height="100" />
</div>

> In order to simplify how our styles are consumed, make it easier to override styles in applications when necessary, and better support those who do not use the Sass pre-processor, we've decided to stop distributing our CSS as SCSS files. We will instead only distribute CSS files and will export all our design tokens as CSS variables. [Have thoughts? Join the discussion on GitHub](https://github.com/CMSgov/design-system/discussions/2223).

Reviewer: ```suggestion
To simplify how our styles are consumed, make it easier to override styles in applications when necessary, and better support those who do not use the Sass pre-processor, we've decided to stop distributing our CSS as SCSS files. We will instead only distribute CSS files and will export all our design tokens as CSS variables. [Have thoughts? Join the discussion on GitHub](https://github.com/CMSgov/design-system/discussions/2223).
```
